design-docs/001-virtual-chunk-improvements.md

```diff
@@ -0,0 +1,163 @@
+# Improvements to virtual chunks handling
+
+## What we need
+
+1. Virtual chunks can be stored in all supported object stores
+2. Multiple credentials sets for virtual chunk resolution
+3. Optimized manifests where the same prefix doesn't need to be repeated for every virtual chunk
+4. Be able to know all the "places" pointed by a repo, without having to load every manifest in every version
+5. Object level change detection (not serving a virtual chunk if the containing file changed)
```

Comment:
I wonder if a more specific requirement here would be helpful. What does "not serve" mean? Is it an error, warning, or configurable behavior?

Reply:
Runtime error when the object has changed instead of getting invalid data

Reply:
Should the Runtime Error be thrown at data-access time or store/session-opening time?

Reply:
It's not practical to do at store-opening time. That would mean checking the etag / modification time of every virtual file eagerly.

Reply:
I changed the description to make it more clear.